src/evaluate.js

```diff
@@ -0,0 +1,379 @@
+      }
+      line('var ' + weightVar(dst) + ' = ' + weightAccessor(readIdx, period) + ';');
+    });
+  }
+
+  if (debug) {
+    if (hasWeights) {
+      line('\n  // Fetch points and project into homogeneous (weighted) coordinates\n');
+    } else {
+      line('\n  // Fetch points\n');
+    }
+  }
+
+  ndloop(n, function (dst) {
+    var readIdx = [];
+    var period = [];
+    for (var d = 0; d < splineDimension; d++) {
+      readIdx[d] = [knotIndex(d), dst[d] - degree[d]];
+      if (boundary[d] === 'closed' && dst[d] - degree[d] < 0) period[d] = sizeVar(d);
+    }
+    if (isBasis) {
+      if (hasWeights) {
+        line('var ' + pointVar(dst) + ' = ' + pointAccessor(readIdx, period) + ' * ' + weightVar(dst) + ';');
+      } else {
+        line('var ' + pointVar(dst) + ' = ' + pointAccessor(readIdx, period) + ';');
+      }
+    } else {
+      for (d = 0; d < spaceDimension; d++) {
+        var dstWithDim = dst.concat(d);
+        readIdx[splineDimension] = d;
+        if (hasWeights) {
+          line('var ' + pointVar(dstWithDim) + ' = ' + pointAccessor(readIdx, period) + ' * ' + weightVar(dst) + ';');
+        } else {
+          line('var ' + pointVar(dstWithDim) + ' = ' + pointAccessor(readIdx, period) + ';');
+        }
+      }
+    }
+  });
+  debugLine('\n');
+
+  debugLine('// Perform De Boor\'s algorithm');
+  for (d = n.length - 1; d >= 0; d--) {
+    n[d] = [degree[d], degree[d] + 1];
+    for (i = 0; i < degree[d]; i++) {
+      debugLine('\n  // Degree ' + degree[d] + ' evaluation in dimension ' + d + ', step ' + (i + 1) + '\n');
+      for (j = degree[d]; j > i; j--) {
+        var isDerivative = derivative && d === derivativeDimension && i === degree[d] - 1;
+
+        if (isDerivative) {
+          line('m = 1 / (' + knotVar([d, j - i + degree[d] - 1]) + ' - ' + knotVar([d, j - 1]) + ');');
+          if (hasWeights) {
+            line('a = (' + tVar(d) + ' - ' + knotVar([d, j - 1]) + ') * m;');
+            line('b = 1 - a;');
+          }
+        } else {
+          line('a = (' + tVar(d) + ' - ' + knotVar([d, j - 1]) + ') / (' + knotVar([d, j - i + degree[d] - 1]) + ' - ' + knotVar([d, j - 1]) + ');');
+          line('b = 1 - a;');
+        }
+
+        if (hasWeights) {
+          ndloop(n, function (ii) {
+            var ij = ii.slice();
+            var ij1 = ii.slice();
+            ij[d] = j;
+            ij1[d] = j - 1;
+            if (isDerivative && hasWeights) line('h = ' + weightVar(ij) + ';');
+            line(weightVar(ij) + ' = b * ' + weightVar(ij1) + ' + a * ' + weightVar(ij) + ';');
+          });
+        }
+        ndloop(n, function (ii) {
+          // Replace the dimension being interpolated with the interpolation indices
+          var ij = ii.slice();
+          var ij1 = ii.slice();
+          ij[d] = j;
+          ij1[d] = j - 1;
+          var ijDim = ij.slice();
+          var ij1Dim = ij1.slice();
```

Comment:
These variable names have become confusing. They can prob be expanded for clarity